src/main/scala/com/tapad/docker/ComposeFile.scala

```diff
@@ -247,6 +255,19 @@ trait ComposeFile extends SettingsHelper with ComposeCustomTagHelpers with Print
     }
   }
 
+  def getStaticPortMappings(ports: Seq[String]): java.util.ArrayList[String] = {
+    val staticPorts = ports.map { port =>
+      val portArray = port.split(":")
+      if (portArray.length == 2) {
+        val portSplitL = { if (portArray(0) == "0") portArray(1) else portArray(0) }
```

Comment:
In fact, you maybe able to reduce all these if/else by using regex along with match/case.